mne/io/bti/bti.py

```diff
@@ -43,6 +43,28 @@
 DTYPES = dict((i, np.dtype(t)) for i, t in dtypes)
 
 
+class bytes_io_mock_context():
+
+    def __init__(self, target):
+        self.target = target
+
+    def __enter__(self):
+        return self.target
+
+    def __exit__(self, type, value, tb):
+        pass
+
+
+def bti_open(fname, *args, **kwargs):
+    """Handle bytes io"""
+    if isinstance(fname, six.string_types):
+        return open(fname, *args, **kwargs)
+    elif isinstance(fname, six.BytesIO):
+        return bytes_io_mock_context(fname)
+    else:
+        raise RuntimeError('Cannot mock this.')
```

Comment:
To get the BytesIo pass the with and open blocks we need to hack a bit as you see.